homework/index.js

```diff
@@ -30,15 +14,119 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function removeChildElements(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+
+  function displayError(error, nodeToDisplay) {
+    removeChildElements(nodeToDisplay);
+    createAndAppend('p', nodeToDisplay, { text: error.message, class: 'alert-error' });
+  }
+
+  function showContributors(contributors) {
+    const contributorsContainer = document.getElementById('contributors');
+    removeChildElements(contributorsContainer);
+    if (contributors) {
+      const ul = createAndAppend('ul', contributorsContainer);
+      for (const contributor of contributors) {
+        const li = createAndAppend('li', ul, { class: 'boxes' });
+        createAndAppend('img', li, { src: contributor.avatar_url, class: 'avatar' });
+        createAndAppend('a', li, {
+          text: contributor.login,
+          href: contributor.html_url,
+          class: 'contributor',
+          target: '_blank',
+        });
+        createAndAppend('div', li, {
+          text: contributor.contributions,
+          class: 'contribution-count',
+        });
       }
+    } else {
+      createAndAppend('p', contributorsContainer, {
+        text: 'No Contributors',
+        class: 'no-contributor',
+      });
+    }
+  }
+
+  function showRepositoryDetails(data) {
+    const repositoryContainer = document.getElementById('repository');
+    removeChildElements(repositoryContainer);
+    const selected = document.getElementById('repositorySelect').selectedIndex;
+    const repository = data[selected];
+    const h2 = createAndAppend('h2', repositoryContainer, { id: 'repositoryName' });
+    createAndAppend('a', h2, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
     });
+    const ul = createAndAppend('ul', repositoryContainer, { id: 'repositoryInfo' });
+    [
+      'Description',
+      repository.description,
+      'Forks',
+      repository.forks_count,
+      'Stargazers',
+      repository.stargazers_count,
+      'Watchers',
+      repository.watchers_count,
+      'Updated',
+      repository.updated_at,
+    ].forEach(value => createAndAppend('li', ul, { text: value }));
+
+    const contributorsContainer = document.getElementById('contributors');
+    fetch(repository.contributors_url)
+      .then(response => response.json())
+      .then(result => showContributors(result))
+      .catch(error => displayError(error, contributorsContainer));
+  }
+
+  function createHeader(data) {
+    const nav = createAndAppend('nav', root, { id: 'navigation', class: 'alignCenter' });
```

Comment:
A small discrepancy here as well. You are using `kebab-casing` for class names above like:
```
        createAndAppend('div', li, {
          text: contributor.contributions,
          class: 'contribution-count',
        });
```
but you use camelCasing here 🙂